docs/security/authentication.md

```diff
@@ -23,6 +23,122 @@ Authentication is company-specific.
+            rewrite ^ https://$host$request_uri? permanent;        # optional, force to use HTTPS
+        }
+
+        location / {
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header Host $http_host;
+            proxy_set_header X-NginX-Proxy true;
+            proxy_pass http://zeppelin;
+            proxy_redirect off;
+            auth_basic "Restricted";
+            auth_basic_user_file /etc/nginx/.htpasswd;
+        }
+    }
+
+    # Zeppelin Websocket
+    server {
+        listen [YOUR-ZEPPELIN-WEBSOCKET-PORT] ssl;    # add ssl is optional, to serve HTTPS connection
+        server_name [YOUR-ZEPPELIN-SERVER-HOST];    # for example: zeppelin.mycompany.com
+
+        ssl_certificate [PATH-TO-YOUR-CERT-FILE];            # optional, to serve HTTPS connection
+        ssl_certificate_key [PATH-TO-YOUR-CERT-KEY-FILE];    # optional, to serve HTTPS connection
+
+        location / {
+            proxy_pass http://zeppelin-wss;
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade websocket;
+            proxy_set_header Connection upgrade;
+            proxy_read_timeout 86400;
+        }
+    }
+    ```
+
+    Then make a symbolic link to this file from `/etc/nginx/sites-enabled/` to enable configuration above when NGINX reloads.
+    
+    ```
+    $ ln -s /etc/nginx/sites-enabled/my-basic-auth /etc/nginx/sites-available/my-basic-auth
+    ```
+
+1. Setup user credential into `.htpasswd` file and restart server
+
+    Now you need to setup `.htpasswd` file to serve list of authenticated user credentials for NGINX server.
+
+    ```
+    $ cd /etc/nginx
+    $ htpasswd -c htpasswd [YOUR_ID]
+    $ NEW passwd: [YOUR_PASSWORD]
+    $ RE-type new passwd: [YOUR_PASSWORD_AGAIN]
+    ```
+    Or you can use your own apache `.htpasswd` files in other location by setup property `auth_basic_user_file`
+
+    Restart NGINX server.
+
+    ```
+    $ service nginx restart
+    ```
+    Then check HTTP Basic Authentication works in browser. If you can see regular basic auth popup and then able to login with credential you entered into `.htpasswd` you are good to go.
+
+    <img src="/assets/themes/zeppelin/img/screenshots/authentication-basic-auth-nginx-request.png" />
+    <img src="/assets/themes/zeppelin/img/screenshots/authentication-basic-auth-nginx-https.png" />
+
+1. More security consideration
+
+* Using HTTPS connection with Basic Authentication is highly recommended since basic auth without encryption may expose your important credential information over the network.
+* Using [Shiro Security feature built-into Zeppelin](https://github.com/apache/incubator-zeppelin/pull/53) is recommended if you prefer all-in-one solution for authentication but NGINX may provides ad-hoc solution for re-use authentication served by your system's NGINX server or in case of you need to separate authentication from zeppelin server.
```

Comment:
Thanks for very useful document.
I think `https://github.com/apache/incubator-zeppelin/blob/master/SECURITY-README.md` is better address for link to shiro security feature.